pkg/scheduler/framework/plugins/podfitshostports/pod_fits_host_ports_test.go

```diff
@@ -0,0 +1,163 @@
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.1/8080")),
+			name:       "same tcp port",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "TCP/127.0.0.1/8080"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.2/8080")),
+			name: "different host ip",
+		},
+		{
+			pod: newPod("m1", "UDP/127.0.0.1/8080"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.1/8080")),
+			name: "different protocol",
+		},
+		{
+			pod: newPod("m1", "UDP/127.0.0.1/8000", "UDP/127.0.0.1/8080"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "UDP/127.0.0.1/8080")),
+			name:       "second udp port conflict",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "TCP/127.0.0.1/8001", "UDP/127.0.0.1/8080"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.1/8001", "UDP/127.0.0.1/8081")),
+			name:       "first tcp port conflict",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "TCP/0.0.0.0/8001"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.1/8001")),
+			name:       "first tcp port conflict due to 0.0.0.0 hostIP",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "TCP/10.0.10.10/8001", "TCP/0.0.0.0/8001"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/127.0.0.1/8001")),
+			name:       "TCP hostPort conflict due to 0.0.0.0 hostIP",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "TCP/127.0.0.1/8001"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/0.0.0.0/8001")),
+			name:       "second tcp port conflict to 0.0.0.0 hostIP",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+		{
+			pod: newPod("m1", "UDP/127.0.0.1/8001"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/0.0.0.0/8001")),
+			name: "second different protocol",
+		},
+		{
+			pod: newPod("m1", "UDP/127.0.0.1/8001"),
+			nodeInfo: schedulernodeinfo.NewNodeInfo(
+				newPod("m1", "TCP/0.0.0.0/8001", "UDP/0.0.0.0/8001")),
+			name:       "UDP hostPort conflict due to 0.0.0.0 hostIP",
+			wantStatus: framework.NewStatus(framework.Unschedulable, predicates.ErrPodNotFitsHostPorts.GetReason()),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			meta := predicates.GetPredicateMetadata(test.pod, nil)
+			state := framework.NewCycleState()
+			state.Write(migration.PredicatesStateKey, &migration.PredicatesStateData{Reference: meta})
```

Comment:
the predicate doesn't read metadata or state, you could remove all three lines above and pass nil to Filter as state.

Reply:
done